influxdb_iox/src/commands/run/main.rs

```
@@ -0,0 +1,173 @@
+use tokio_util::sync::CancellationToken;
+
+#[cfg(all(not(feature = "heappy"), feature = "jemalloc_replacing_malloc"))]
+mod jemalloc;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("{}", source))]
+    Wrapper { source: ioxd_common::Error },
+
+    #[snafu(display("Error joining server task: {}", source))]
+    Joining { source: tokio::task::JoinError },
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+impl From<ioxd_common::Error> for Error {
+    fn from(source: ioxd_common::Error) -> Self {
+        Self::Wrapper { source }
+    }
+}
+
+#[cfg(all(not(feature = "heappy"), not(feature = "jemalloc_replacing_malloc")))]
+fn build_malloc_conf() -> String {
+    "system".to_string()
+}
+
+#[cfg(all(feature = "heappy", not(feature = "jemalloc_replacing_malloc")))]
+fn build_malloc_conf() -> String {
+    "heappy".to_string()
+}
+
+#[cfg(all(not(feature = "heappy"), feature = "jemalloc_replacing_malloc"))]
+fn build_malloc_conf() -> String {
+    tikv_jemalloc_ctl::config::malloc_conf::mib()
+        .unwrap()
+        .read()
+        .unwrap()
+        .to_string()
+}
+
+#[cfg(all(
+    feature = "heappy",
+    feature = "jemalloc_replacing_malloc",
+    not(feature = "clippy")
+))]
+fn build_malloc_conf() -> String {
+    compile_error!("must use exactly one memory allocator")
+}
+
+#[cfg(feature = "clippy")]
+fn build_malloc_conf() -> String {
+    "clippy".to_string()
+}
+
+/// This is the entry point for the IOx server.
+///
+/// This entry point ensures that the given set of Services are
+/// started using best practice, e.g. that we print the GIT-hash and
+/// malloc-configs, that a panic handler is installed, etc.
+///
+/// Due to its invasive nature (install global panic handling,
+/// logging, etc) this function should not be used during unit tests.
+pub async fn main(common_state: CommonServerState, services: Vec<Service>) -> Result<()> {
+    let git_hash = env!("GIT_HASH", "starting influxdb_iox server");
```

The change is to use `env!` here rather than `option_env!` to fail build if githash is not set